### PR TITLE
Rewrite Flow Refactor

### PR DESCRIPTION
**Pull request by waigani**
- Refactor to parse decorator args like cli input, extracting flags and args.
- Generalise the Flow SDK. Use the updated SDK to refactor the Rewrite Flow.
Other Flows should now be able to reuse the SDK.

### CODE REVIEW

File: sdk/flow/cli.go [hunk 1]
```diff
@@ -3,38 +3,175 @@ package flow
...
+	}
+
+	// If user is manually confirming reviews, set a long timeout.
+	timeout := time.After(time.Hour * 1)
+
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	go func() {
+		wg.Wait()
+		close(decoratedResultc)
```
Comment (BlakeMScurr) [inline sdk/flow/cli.go]: Won't this panic, because the channel is already closed in a defer at the top of the function?

File: sdk/flow/cli.go [hunk 2]
```diff
@@ -3,38 +3,175 @@ package flow
...
+			close(decoratedResultc)
+		}
+	}()
+
+	resultc, errc, cancel, err := f.RunCLI()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	// If user is manually confirming reviews, set a long timeout.
```
Comment (BlakeMScurr) [inline sdk/flow/cli.go]: Reference to "review" should be generic.

File: sdk/flow/cli.go [hunk 3]
```diff
@@ -3,38 +3,175 @@ package flow
...
+l:
+	for {
+		select {
+		case err, ok := <-errc:
+			if !ok {
+				errc = nil
+				break
+			}
+
+			util.Logger.Debugf("Review error: %s", errors.ErrorStack(err))
```
Comment (BlakeMScurr) [inline sdk/flow/cli.go]: "review"

File: sdk/flow/cli.go (hunk 1 above)
Comment (waigani) [reply sdk/flow/cli.go]: Only if we hit an error. A better pattern is to use a wait chan, I'll update it.

**Pull request merged**